resource-usage-collect-addon/pkg/addon/agent/calculate.go

```diff
@@ -2,120 +2,174 @@ package agent
+	availableTpu := tpuAlloc - tpuUsage
+	tpuScoreNormalizer := clustersdkv1alpha1.NewScoreNormalizer(MINTPUCOUNT, MAXTPUCOUNT)
+	tpuScore, err = tpuScoreNormalizer.Normalize(availableTpu)
+
+	klog.Infof("cpuScore = %v, memScore = %v, gpuScore = %v, tpuScore = %v", cpuScore, memScore, gpuScore, tpuScore)
+	return cpuScore, memScore, gpuScore, tpuScore, nil
 }
 
-func (s *Score) calculateClusterAllocateable(resourceName clusterv1.ResourceName) (float64, error) {
+// Iterate every node, find the node with maximum allocatable resource, return the number and node name.
+func (s *Score) calculateClusterAllocatable(resourceName string) (float64, string, error) {
 	nodes, err := s.nodeLister.List(labels.Everything())
 	if err != nil {
-		return 0, err
+		return 0, "", err
 	}
 
-	allocatableList := make(map[clusterv1.ResourceName]resource.Quantity)
+	var maxAllocatable float64
+	var maxNodeName string
 	for _, node := range nodes {
 		if node.Spec.Unschedulable {
 			continue
 		}
-		for key, value := range node.Status.Allocatable {
-			if allocatable, exist := allocatableList[clusterv1.ResourceName(key)]; exist {
-				allocatable.Add(value)
-				allocatableList[clusterv1.ResourceName(key)] = allocatable
-			} else {
-				allocatableList[clusterv1.ResourceName(key)] = value
-			}
+		alloc, exists := node.Status.Allocatable[v1.ResourceName(resourceName)]
+		if !exists {
+			continue
+		}
+		klog.Infof("Node: %s, Allocatable %s: %f", node.Name, resourceName, alloc.AsApproximateFloat64())
+		if alloc.AsApproximateFloat64() > maxAllocatable {
+			maxAllocatable = alloc.AsApproximateFloat64()
+			maxNodeName = node.Name
 		}
 	}
-	quantity := allocatableList[resourceName]
-	return quantity.AsApproximateFloat64(), nil
+	klog.Infof("Max allocatable %s: %f on node: %s", resourceName, maxAllocatable, maxNodeName)
+	return maxAllocatable, maxNodeName, nil
 }
 
-func (s *Score) calculatePodResourceRequest(resourceName v1.ResourceName) (float64, error) {
-	list, err := s.podListener.List(labels.Everything())
+func (s *Score) getRequestForResource(resource string, requests *v1.ResourceList, nonZero bool) float64 {
+	if requests == nil {
+		return 0
+	}
+	switch resource {
+	case string(v1.ResourceCPU):
+		// Override if un-set, but not if explicitly set to zero
+		if _, found := (*requests)[v1.ResourceCPU]; !found && nonZero {
+			return 100
+		}
+		return requests.Cpu().AsApproximateFloat64()
+	case string(v1.ResourceMemory):
+		// Override if un-set, but not if explicitly set to zero
+		if _, found := (*requests)[v1.ResourceMemory]; !found && nonZero {
+			return 200 * 1024 * 1024
+		}
+		return requests.Memory().AsApproximateFloat64()
+	default:
+		quantity, found := (*requests)[v1.ResourceName(resource)]
+		if !found {
+			return 0
+		}
+		return quantity.AsApproximateFloat64()
+	}
+}
+
+func (s *Score) calculateNodeResourceUsage(nodeName string, resourceName string) (float64, error) {
+	list, err := s.podLister.List(labels.Everything())
```

Comment:
`Completed` pods should be filtered out. Eg,
```
NAME                   READY   STATUS      RESTARTS   AGE
demo1-job8n84t-bgzhs   0/1     Completed   0          19m
```

Reply:
I have fixed that